iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/BusAdapter.java

```diff
@@ -19,28 +23,48 @@
 class BusAdapter extends AbstractIdentifiableAdapter<Bus> implements Bus {
 
     private final List<TerminalAdapter> terminals = new ArrayList<>();
+    private final TerminalAdapter terminalRef;
 
     BusAdapter(final Bus delegate, final MergingViewIndex index) {
         super(delegate, index);
 
-        delegate.visitConnectedEquipments(new AbstractTerminalTopologyVisitor() {
-            @Override
-            public void visitTerminal(Terminal t) {
-                terminals.add(getIndex().getTerminal(t));
-            }
-        });
+        delegate.getConnectedTerminalStream()
+                .map(t -> getIndex().getTerminal(t))
+                .forEach(terminals::add);
+        if (terminals.isEmpty() && delegate.getVoltageLevel().getTopologyKind() == TopologyKind.NODE_BREAKER) {
+            // If we are in this case, the bus in a calculated Bus-Breaker bus which is between two retained switches
+            terminalRef = findTerminalRef(delegate, getIndex(), (s, v) -> v.getBusBreakerView().getBus1(s.getId()), (s, v) -> v.getNodeBreakerView().getNode1(s.getId()))
+                    .orElseGet(() ->
+                            findTerminalRef(delegate, getIndex(), (s, v) -> v.getBusBreakerView().getBus2(s.getId()), (s, v) -> v.getNodeBreakerView().getNode2(s.getId()))
+                                    .orElse(null));
+        } else {
+            terminalRef = terminals.isEmpty() ? null : terminals.get(0);
+        }
+    }
+
+    private static Optional<TerminalAdapter> findTerminalRef(Bus delegate, MergingViewIndex index,
+                                                             BiFunction<Switch, VoltageLevel, Bus> busGetter, BiFunction<Switch, VoltageLevel, Integer> nodeGetter) {
+        VoltageLevel vl = delegate.getVoltageLevel();
+        return vl.getNodeBreakerView().getSwitchStream()
+                .filter(Switch::isRetained)
+                .filter(s -> busGetter.apply(s, vl).equals(delegate))
+                .map(s -> nodeGetter.apply(s, vl))
+                .map(i -> Networks.getEquivalentTerminal(vl, i))
+                .filter(Objects::nonNull)
+                .map(index::getTerminal)
+                .findFirst();
```

Comment:
Why don't we use the same implementation than in the `CalculatedBusImpl`?

Honestly, I don't understand whitch criterias you used to chose the reference terminal. I think you should comment a little bit the algorithm.

Reply:
It was because in merging view, we don't have the nodes included in the calculated bus (which are used in network-impl). However, I agree with you that implementing the algorithm a second time here to find the nodes is error-prone. I put the `getTerminalReference()` method in the code which simplify the code A LOT. The default implementation is not exactly correct but it is close enough and this way, we use the algorithm directly from `CalculatedBusImpl`.